tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -199,6 +199,25 @@ public function testColumnCollation()
         $this->assertEquals('utf8_general_ci', $columns['bar']->getPlatformOption('collation'));
     }
 
+    public function testColumnCharset()
+    {
+        $table = new Table('test_charset');
+        $table->addOption('collate', $collation = 'latin1_swedish_ci');
+        $table->addOption('charset', 'latin1');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('text', 'text');
+        $table->addColumn('foo', 'text')->setPlatformOption('charset', 'latin1');
+        $table->addColumn('bar', 'text')->setPlatformOption('charset', 'utf8');
+        $this->_sm->dropAndCreateTable($table);
+
+        $columns = $this->_sm->listTableColumns('test_charset');
+
+        $this->assertArrayNotHasKey('charset', $columns['id']->getPlatformOptions());
+        $this->assertEquals('latin1', $columns['text']->getPlatformOption('charset'));
```

Comment:
Yes, I've set the default table charset to `latin1` : https://github.com/doctrine/dbal/pull/881/files#diff-921db63349be598a376afb1584ccc8b5R206, the test is verifying the default charset behavior.